src/components/Footer.tsx

```diff
@@ -0,0 +1,85 @@
+import Divider from '@mui/material/Divider'
+import Link from 'next/link'
+import { FaFacebookF, FaInstagram, FaTiktok, FaYoutube } from 'react-icons/fa'
+import { FaXTwitter } from 'react-icons/fa6'
+import useWindowDimensions from '~/hooks/useWindowDimensions'
+
+const Footer = () => {
+	const { height, width } = useWindowDimensions()
```

Comment:
I couldn't use this hook because I had a lot of problems with mismatched hydration, because in next.js everything is rendered on the server first and then on the client. There is workaround using useEffect probably, but I choose CSS
![Zrzut ekranu 2024-09-21 134812](https://github.com/user-attachments/assets/cf548f20-ce39-4817-bd74-3ff048d17894)
![Zrzut ekranu 2024-09-21 134802](https://github.com/user-attachments/assets/c837ea46-4304-48a4-80f0-c7066cb17276)
![Zrzut ekranu 2024-09-21 134747](https://github.com/user-attachments/assets/705263bd-045b-4599-a6be-2cad2e3b93ba)

Reply:
Yeah, there're at least few solutions to this issue:
1. `use client` directive (the easiest one)
2. Create a component (eg. Media):
```
import type { Theme } from '@emotion/react'
import { css } from '@emotion/react'
import styled from '@emotion/styled'

type Breakpoints = keyof Theme[`breakpoints`][`values`]
type BreakpointProperty = Breakpoints | number

export const isNumeric = (value: string | number): value is number => typeof value === `number`

export interface MediaProps {
  at?: BreakpointProperty
  lessThan?: BreakpointProperty
  greaterThan?: BreakpointProperty
  greaterThanOrEqual?: BreakpointProperty
  between?: [BreakpointProperty, BreakpointProperty]
}

export interface WrapperProps {
  at?: MediaProps[`at`]
  lessThan?: MediaProps[`lessThan`]
  greaterThan?: MediaProps[`greaterThan`]
  greaterThanOrEqual?: MediaProps[`greaterThanOrEqual`]
  between?: MediaProps[`between`]
  breakpointsKeys: Breakpoints[]
}

export const Wrapper = styled.div<WrapperProps>`
  width: inherit;
  ${({ theme, at }) =>
    at &&
    css`
      display: none;

      ${theme.breakpoints.only(at)} {
        display: unset;
      }
    `}

  ${({ theme, breakpointsKeys, lessThan }) =>
    lessThan &&
    css`
      display: none;

      ${theme.breakpoints.down(
        isNumeric(lessThan) ? lessThan : breakpointsKeys[breakpointsKeys.indexOf(lessThan) - 1]
      )} {
        display: unset;
      }
    `}

  ${({ theme, breakpointsKeys, greaterThan }) =>
    greaterThan &&
    css`
      display: none;

      ${theme.breakpoints.up(
        isNumeric(greaterThan) ? greaterThan : breakpointsKeys[breakpointsKeys.indexOf(greaterThan) + 1]
      )} {
        display: unset;
      }
    `}
 
 ${({ theme, greaterThanOrEqual }) =>
    greaterThanOrEqual &&
    css`
      display: none;

      ${theme.breakpoints.up(greaterThanOrEqual)} {
        display: unset;
      }
    `}

${({ theme, between }) =>
    between &&
    css`
      display: none;

      ${theme.breakpoints.between(...between)} {
        display: unset;
      }
    `}
`



import type { FC, PropsWithChildren } from 'react'
import { useMemo, memo } from 'react'
import { useTheme } from '@emotion/react'

import * as views from './Media.views'

type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
  {
    [K in Keys]-?: Required<Pick<T, K>> & Partial<Pick<T, Exclude<Keys, K>>>
  }[Keys]

export const Media: FC<PropsWithChildren<RequireAtLeastOne<views.MediaProps>>> = memo(
  ({ at, lessThan, greaterThan, greaterThanOrEqual, between, children }) => {
    const { breakpoints } = useTheme()
    const breakpointsKeys = Object.keys(breakpoints.values) as (keyof typeof breakpoints.values)[]

    if (greaterThan === breakpointsKeys[breakpointsKeys.length - 1]) {
      throw new Error(`There is no breakpoint larger than ${greaterThan}`)
    }

    const queries = useMemo(
      () => ({
        at: at
          ? breakpoints.between(at, views.isNumeric(at) ? at : breakpointsKeys[breakpointsKeys.indexOf(at)], false)
          : ``,
        lessThan: lessThan ? breakpoints.down(lessThan, false) : ``,
        greaterThan: greaterThan
          ? breakpoints.up(
              views.isNumeric(greaterThan) ? greaterThan : breakpointsKeys[breakpointsKeys.indexOf(greaterThan) + 1],
              false
            )
          : ``,
        greaterThanOrEqual: greaterThanOrEqual ? breakpoints.up(greaterThanOrEqual, false) : ``,
        between: between ? breakpoints.between(...between, false) : ``,
      }),
      [at, between, breakpoints, breakpointsKeys, greaterThan, greaterThanOrEqual, lessThan]
    )

    const isAtResolution = useMatchMedia(queries.at, true)
    const isLessThanResolution = useMatchMedia(queries.lessThan, true)
    const isGreaterThanResolution = useMatchMedia(queries.greaterThan, true)
    const isGreaterThanOrEqualResolution = useMatchMedia(queries.greaterThanOrEqual, true)
    const isBetweenResolution = useMatchMedia(queries.between, true)

    return (
      <views.Wrapper
        at={at}
        lessThan={lessThan}
        greaterThan={greaterThan}
        greaterThanOrEqual={greaterThanOrEqual}
        between={between}
        breakpointsKeys={breakpointsKeys}
        data-testid="media-wrapper"
      >
        {(isAtResolution ||
          isLessThanResolution ||
          isGreaterThanResolution ||
          isGreaterThanOrEqualResolution ||
          isBetweenResolution) &&
          children}
      </views.Wrapper>
    )
  }
)
```

This one will render both variants on the server site and on the client site will remove the redundant one from the DOM.

3. Use the ready-to-use libs like [Fresnel](https://github.com/artsy/fresnel)